### PR TITLE
[Silabs] Refactor SiWx917 Wi-Fi Application implementation

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description 
PR transitions the SiWx917 Wi-Fi implementation to use the sl_net API. This allows us to leverage the LwIP integration of the Wi-Fi SDK instead of using the one implemented for Matter for the SiWx917.

- Replace sl_wifi APIs by the sl_net equivalent
- Remove the join callback handler since the sl_net_up API does not trigger the callback and that it is not necessary anymore
- Remove the scan network callback since it always failed due to string length issues. Since we assume the scan can fail and still try to connect, the scan brings little value.

#### Tests
Siw917x build, flash and commission

### CODE REVIEW

File: examples/platform/silabs/MatterConfig.cpp [hunk 1]
```diff
@@ -185,13 +185,13 @@ void SilabsMatterConfig::AppInit()
 {
     GetPlatform().Init();
     sMainTaskHandle = osThreadNew(ApplicationStart, nullptr, &kMainTaskAttr);
-    SILABS_LOG("Starting scheduler");
+    ChipLogProgress(DeviceLayer, "Starting scheduler");
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/MatterConfig.cpp:188]: Why do you choose Appserver for some and DeviceLayer for others here?

File: examples/platform/silabs/SiWx917/sl_wifi_if.cpp [hunk 2]
```diff
@@ -69,58 +77,126 @@ extern "C" {
...
+                                                 ),
+                     .config_feature_bit_map = (SL_SI91X_FEAT_SLEEP_GPIO_SEL_BITMAP | RSI_CONFIG_FEATURE_BITMAP) }
+};
+
+constexpr int8_t kAdvScanThreshold           = -40;
+constexpr uint8_t kAdvRssiToleranceThreshold = 5;
+constexpr uint8_t kAdvActiveScanDuration     = 15;
+constexpr uint8_t kAdvPassiveScanDuration    = 20;
+constexpr uint8_t kAdvMultiProbe             = 1;
+constexpr uint8_t ADV_SCAN_PERIODICITY       = 10;
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/SiWx917/sl_wifi_if.cpp:184]: does it have to remain named like a define?

File: examples/platform/silabs/SiWx917/sl_wifi_if.cpp [hunk 3]
```diff
@@ -152,6 +228,180 @@ static void StartDHCPTimer(uint32_t timeout)
...
+                        ChipLogError(DeviceLayer, "sl_wifi_get_firmware_version failed: 0x%lx", static_cast<uint32_t>(status)));
+
+    ChipLogDetail(DeviceLayer, "Firmware version is: %x%x.%d.%d.%d.%d.%d.%d", version.chip_id, version.rom_id, version.major,
+                  version.minor, version.security_version, version.patch_num, version.customer_id, version.build_num);
+
+    status = sl_wifi_get_mac_address(SL_WIFI_CLIENT_INTERFACE, (sl_mac_address_t *) &wfx_rsi.sta_mac.octet[0]);
+    VerifyOrReturnError(status == SL_STATUS_OK, status,
+                        ChipLogError(DeviceLayer, "sl_wifi_get_mac_address failed: 0x%lx", static_cast<uint32_t>(status)));
+
+#ifdef SL_MBEDTLS_USE_TINYCRYPT
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/SiWx917/sl_wifi_if.cpp:277]: can we make it always defined?

File: examples/platform/silabs/SiWx917/sl_wifi_if.cpp [hunk 4]
```diff
@@ -788,34 +693,34 @@ void ProcessEvent(WfxEvent_t inEvent)
...
     case WFX_EVT_STA_DISCONN:
         ChipLogDetail(DeviceLayer, "WFX_EVT_STA_DISCONN");
         // TODO: This event is not being posted anywhere, seems to be a dead code or we are missing something
         wfx_rsi.dev_state &=
             ~(WFX_RSI_ST_STA_READY | WFX_RSI_ST_STA_CONNECTING | WFX_RSI_ST_STA_CONNECTED | WFX_RSI_ST_STA_DHCP_DONE);
         /* TODO: Implement disconnect notify */
         ResetDHCPNotificationFlags();
-        wfx_lwip_set_sta_link_down(); // Internally dhcpclient_poll(netif) ->
-                                      // wfx_ip_changed_notify(0) for IPV4
+        // wfx_lwip_set_sta_link_down(); // Internally dhcpclient_poll(netif) ->
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/SiWx917/sl_wifi_if.cpp:706]: Why isn't needed anymore? Why do not delete it

File: examples/platform/silabs/efr32/rs911x/rsi_if.cpp [hunk 5]
```diff
@@ -344,31 +344,31 @@ static void wfx_rsi_wlan_pkt_cb(uint16_t status, uint8_t * buf, uint32_t len)
...
  *****************************************************************************************/
-static int32_t wfx_rsi_init(void)
+static int32_t sl_matter_wifi_init(void)
 {
     int32_t status;
     uint8_t buf[RSI_RESPONSE_HOLD_BUFF_SIZE];
     extern void rsi_hal_board_init(void);
 
-    ChipLogProgress(DeviceLayer, "wfx_rsi_init: starting(HEAP_SZ = %d)", SL_HEAP_SIZE);
+    ChipLogProgress(DeviceLayer, "sl_matter_wifi_init: starting(HEAP_SZ = %d)", SL_HEAP_SIZE);
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/efr32/rs911x/rsi_if.cpp:360]: do we need all the `sl_matter_wifi_init` in the traces here?

File: src/platform/silabs/SiWx917/wifi/ethernetif.cpp [hunk 6]
```diff
@@ -72,14 +68,13 @@ uint32_t gOverrunCount = 0;
  * @return
  *    None
  ******************************************************************************/
-static void low_level_init(struct netif * netif)
+void low_level_init(struct netif * netif)
```
Comment (jmartinez-silabs) [inline src/platform/silabs/SiWx917/wifi/ethernetif.cpp:71]: didn't all this exist in gsdk too? I thought we reviews this last week

File: src/platform/silabs/SiWx917/wifi/lwip_netif.cpp [hunk 7]
```diff
@@ -123,7 +140,9 @@ void wfx_lwip_set_sta_link_down(void)
 void wfx_lwip_start(void)
 {
     /* Initialize the LwIP stack */
-    netif_config(&sta_netif, NULL);
+    // netif_config(&sta_netif, NULL);
```
Comment (jmartinez-silabs) [inline src/platform/silabs/SiWx917/wifi/lwip_netif.cpp:143]: cleanup

File: src/platform/silabs/SiWx917/wifi/ethernetif.cpp (hunk 6 above)
Comment (mkardous-silabs) [reply src/platform/silabs/SiWx917/wifi/ethernetif.cpp:71]: the APIs are in the wiseconnect sdk so i'm not sure if they work for the 9116 or the wf200 which is why i left them for now.

File: examples/platform/silabs/MatterConfig.cpp (hunk 1 above)
Comment (mkardous-silabs) [reply examples/platform/silabs/MatterConfig.cpp:188]: Arbitrary choice - i wasn't sure which i should take

File: examples/platform/silabs/SiWx917/sl_wifi_if.cpp (hunk 2 above)
Comment (mkardous-silabs) [reply examples/platform/silabs/SiWx917/sl_wifi_if.cpp:184]: fixed

File: src/platform/silabs/SiWx917/wifi/lwip_netif.cpp (hunk 7 above)
Comment (mkardous-silabs) [reply src/platform/silabs/SiWx917/wifi/lwip_netif.cpp:143]: file was delete

File: examples/platform/silabs/SiWx917/sl_wifi_if.cpp (hunk 4 above)
Comment (mkardous-silabs) [reply examples/platform/silabs/SiWx917/sl_wifi_if.cpp]: removed

File: examples/platform/silabs/efr32/rs911x/rsi_if.cpp (hunk 5 above)
Comment (mkardous-silabs) [reply examples/platform/silabs/efr32/rs911x/rsi_if.cpp]: removed

**Pull request closed without merging**